test/mitmproxy/console/test_master.py

```diff
@@ -37,6 +46,13 @@ def test_basic(self):
             self.dummy_cycle(m, 1, b"")
             assert len(m.view) == i
 
+    @mock.patch('mitmproxy.tools.console.signals.add_log', side_effect=mock_add_log)
+    def test_run_script_once(self, test_func):
+        m = self.mkmaster()
+        f = tflow.tflow(resp=True)
+        with mitmproxy.test.tutils.raises(ScriptError):
```

Comment:
Have you considered just running `m.run_script_once("nonexistent", [f])` and then checking `m.logbuffer` ?